src/kOS.Safe/Communication/Connection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+
+namespace kOS.Safe.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection")]
+    public abstract class Connection<S> : Structure where S : SharedObjects
```

Comment:
If you just tell it to use the kOS.Safe.SharedObjects class wouldn't it still allow you to assign a kOS.SharedObjects into it just fine?
Using the 'generics' syntax basically means you want there to be able to exist more than one variation of the class.  It means you want there to be two *different kinds* of `Connection` - one that can work on kOS.Safe.SharedObjects and a *different one* that can work on kOS.SharedObjects.  Is there a reason you have to make two versions of the Connection class, one for use with kOS.Safe and one for use with kOS (unsafe), or will it work just fine to have only one version, using the kOS.Safe version, and then have the kOS (unsafe) version just use that directly (it should be perfectly happy with letting you store an (unsafe) SharedObjects in a member designated as a kOS.Safe.SharedObjects)?